library/std/src/os/fd/raw.rs

```diff
@@ -207,6 +214,7 @@ impl AsRawFd for io::Stderr {
 }
 
 #[stable(feature = "asraw_stdio_locks", since = "1.35.0")]
+#[cfg(not(target_os = "trusty"))]
```

Comment:
Adding a bunch of `cfg` seems like a tacit admission that this implementation is not very useful, and depends on certain implementations never being even referenced.

In general, we have avoided adding `cfg` to the implementations of std. I do not think we should get in the habit here.

Reply:
Is there a better approach to follow for disabling unsupported parts of std? Would it be better to instead stub out platform APIs with `unimplemented!`? That would involve fewer `cfg`s, though I'm not sure how to handle tests with that approach.

Reply:
There is really no satisfactory answer, to be clear, which is why we have shifted towards being increasingly reluctant to allow targets to add only partial support for std.